lib/runner.js

```diff
@@ -229,7 +229,7 @@ Runner.prototype.fail = function (test, err) {
   ++this.failures;
   test.state = 'failed';
 
-  if (!(err instanceof Error || err && typeof err.message === 'string')) {
+  if (!((err instanceof Error || err) && typeof err.message === 'string')) {
```

Comment:
Are we sure this shouldn't be `if (!(err instanceof Error || (err && typeof err.message === 'string'))) {`? Reading the original, and not remembering exactly how JS evaluates differing chained boolean operators without parentheses, it *looks* like the original included `err` as a condition to guard against accessing `.message` of `undefined`.

Reply:
I think you're right.